application/src/main/webapp/js/chat.js

```diff
@@ -1648,17 +1659,41 @@ ChatApplication.prototype.getUsers = function(roomId, callback, asString) {
  * @param filter : the filter (ex: Ben Pa)
  * @param callback : return the json users data list as a parameter of the callback function
  */
-ChatApplication.prototype.getAllUsers = function(filter, callback) {
+ChatApplication.prototype.getAllUsers = function(filter, isSearchInRoomFirst, callback) {
+
+  // Cache result
+  var _this = this;
+  if (this.getAllUsersCache == undefined) {
+    this.getAllUsersCache = {};
+  }
+  var cacheKey = filter;
+  if (isSearchInRoomFirst) {
+    cacheKey += '__' + this.targetUser;
+  }
+  if (this.getAllUsersCache[cacheKey] != undefined) {
```

Comment:
IMO, refresh the page to clear cache in JavaScript is enough for us now. Usually, user does not do this search frequently and the size of cache is not too big. 
Actually, some javascript libraries in PLF work in the same way (as mentioning and assign/coworker in task management)